src/managing-workflow/security-considerations.md

```diff
@@ -1,3 +1,56 @@
 # Security Considerations
 
-TODO (bacongobbler): rewrite for v2
+!!! important
+	Workflow is not suitable for multi-tenant environments or hosting untrusted code.
+
+A major goal of Workflow is to be operationally secure and trusted by operations engineers in every
+deployed environment. There are, however, two notable security-related considerations to be aware
+of when deploying Workflow.
+
+# Application Runtime Segregation
+
+Users of Workflow often want to deploy their applications to separate environments (commonly:
+development, staging, and production). Typically, physical network isolation isn’t the goal, but
+rather segregation of application environments - if a development app goes haywire, it shouldn’t
+affect production applications that are running in the cluster.
+
+In Workflow, deployed applications can be segregated by using the `deis tags` command. This enables
+you to tag machines in your cluster with arbitrary metadata, then configure your applications to be
+scheduled to machines which match the metadata.
+
+For example, if some machines in your cluster are tagged with `environment=production` and some
+with `environment=staging`, you can configure an application to be deployed to the production
+environment by using `deis tags set environment=production`. Workflow will pass this configuration
+along to Kubernetes, which will schedule your application in different environments running on
+separate hardware.
+
+# Running Workflow on Public Clouds
+
+If you are running on a public cloud without security group features, you will have to set up
+security groups yourself through either `iptables` or a similar tool. The only ports on worker
+nodes that should be exposed to the public are:
+
+ - 22: (optional) for remote SSH
+ - 80: for the routers
+ - 443: (optional) routers w/ SSL enabled
+ - 2222: for the routers proxying TCP to the builder
```

Comment:
9090 should be on the list as well.  It's a port used by the router for healthchecks and healthchecks _only_. Healthchecks can still be done using 80 or 443, but 9090 specifically offers a guarantee that it will _never_ use PROXY protocol, even if 80 and 443 _are_ configured to use it. This is important, as it turns out, for an HTTP/S- based healthcheck from some load balancers to actually succeed. You probably don't need to go into that level of detail here, but 9090 does belong on the list.